Argo/Types/Decoded.swift

```diff
@@ -12,14 +12,23 @@ public enum Decoded<T> {
     default: return .None
     }
   }
+}
 
-  public static func optional<A>(p: Decoded<A>) -> Decoded<A?> {
-    switch p {
+public extension Decoded {
+  static func optional<T>(x: Decoded<T>) -> Decoded<T?> {
```

Comment:
I could also go either way on this one. Just out of curiosity, what would you expect a function named `optional` to return?